src/aiori-MPIIO.c

```diff
@@ -650,25 +650,23 @@ IOR_offset_t MPIIO_GetFileSize(aiori_mod_opt_t * module_options, char *testFileN
           return 0;
         IOR_offset_t aggFileSizeFromStat, tmpMin, tmpMax, tmpSum;
         MPI_File fd;
-        MPI_Comm comm;
         MPI_Info mpiHints = MPI_INFO_NULL;
 
-        if (hints->filePerProc == TRUE) {
-                comm = MPI_COMM_SELF;
-        } else {
-                comm = testComm;
+        if (hints->filePerProc || rank == 0) {
```

Comment:
I initially thought we needed a sync/barrier/sync here but GetFileSize is called from CheckFileSize , where we just closed the file and issued a barrier.  that seems pretty safe for even the strangest of strange file systems.